### ISSUE TITLE
Disambiguating colloquialisms

### ISSUE THREAD

**Issue by Public-Health-Bioinformatics** [FoodOntology/foodon#20]
Matthew's point:

> 1. In addition to the above examples, as you mention, we need to be
>    able to model food names to species/higher taxonomic orders. This gets
>    tricky because often the same names for foods refer to different
>    species--depending on locale and culture. This means that these are not
>    just strict translation issues, but ethnic and cultural colloquialisms that
>    should be modeled in from the beginning. Your coverage of crayfish is a
>    good example--there is decent synonymy in LanguaL for crayfish<=>crawfish,
>    but only one reference for "crawdad"--which is the word I used when I was
>    growing up. Even more puzzling is the use of the word "Shrimp" as
>    food...admittedly LanguaL does a decent job trying to parse this out--but
>    if one only has "shrimp" in a recipe, it would be quite difficult to
>    determine what is actually meant.

What I'm hoping is that we can use annotations to disambiguate semantics for homographs/homonyms. Under the hood, FOODON has a unique id for each term, so two or more terms can have the same label but different ids, and can be disambiguated in data storage.  We'll be able to offer different latin scientific name synonyms for respective versions of the term, as well as cultural / bioregional annotations.  Emma tells me of a "wild or African spinach" example which refers to a completely different set of plants (Morogo); I also find "New Zealand spinach", same story (see also https://en.wikipedia.org/wiki/Spinach_(disambiguation).  To encourage disambiguation do we offer multiple concepts with disambiguating name if possible, and offer the homonym as a synonym?  I thought of this kind of mockup:

```
  label: New Zealand spinach
  id: FOODON_351732
  xref: Langual: B1732
  synonym: "Tetragonia tetragonioides (Pallas) Kuntze" NARROW LATIN [ITIS: 19938, NCBI:45318]
  synonym: "Tetragonia expansa" NARROW LATIN
  synonym: "spinach" BROAD
  "has regional usage" [Argentina, Australia, Chile, Japan, New Zealand]

  label: spinach
  xref: Langual: B1420
  id: FOODON_351420
  synonym: "Spinacia oleracea L."   NARROW LATIN [ITIS:20709, NCBI:3562]
  "has regional usage" [North America]

  label: morogo
  id: FOODON_356789
  synonym:"Amaranthus sp." NARROW LATIN [ITIS: 20715, NCBI:3564 ]
  synonym: "African spinach" EXACT
  synonym: "spinach" BROAD
  "has regional usage" "Southern Africa"
```

I guess it can be hard to pick the most common name for a species or group of species.

Does the "has regional usage" sound like a good solution for pairing term context to applicable concept?

About crayfish vs crawfish vs crawdad: For exact synonyms (where one can be substituted for another) FOODON will have to pick one for a class label, and the remainder become EXACT synonym annotations.  In software user interfaces for input of datums ideally the synonyms are consulted during type-as-you-go data entry.  If a synonym is particular cultural dialect, it can be annotated that way.  Any other FOODON people have examples of that though?  I haven't done it before.  This approach only works though if all terms apply to same scientific named examples.  

But there's a problem with the synonym approach when it comes to recipes.  A recipe must keep track of the author's choice of words but it can't do this if only one term id represents all synonym variations taken in by a parser.  One would have to annotate an ingredient with the specific synonym chosen.  That suggests each synonym should be given its own ontology id to concisely capture original data.  Hmm.

**Comment by Public-Health-Bioinformatics** [comment]
One other relation should be established to connect a food source to possible ethnic specificity.  Shall we import the Experimental Factor Ontology list of [ethnic groups](http://www.ontobee.org/ontology/EFO?iri=http://www.ebi.ac.uk/efo/EFO_0001799) for this?

**Comment by Public-Health-Bioinformatics** [comment]
@leechuck comments about:

> 1. In addition to the above examples, as you mention, we need to be
>    able to model food names to species/higher taxonomic orders. This gets
>    tricky because often the same names for foods refer to different
>    species--depending on locale and culture. This means that these are
>    not just strict translation issues, but ethnic and cultural
>    colloquialisms that should be modeled in from the beginning. Your
>    coverage of crayfish is a good example--there is decent synonymy in
>    LanguaL for crayfish<=>crawfish, but only one reference for
>    "crawdad"--which is the word I used when I was growing up. Even more
>    puzzling is the use of the word "Shrimp" as food...admittedly LanguaL
>    does a decent job trying to parse this out--but if one only has
>    "shrimp" in a recipe, it would be quite difficult to determine what is
>    actually meant.

Maybe the following might work: define "shrimp (sensu X)" classes for
each use of shrimp. Then define a general "shrimp" class as the union of
all the individual classes (or even as a superclass of the union, so to
allow the union to be incomplete).

R

**Comment by Public-Health-Bioinformatics** [comment]
@mateolan about "shrimp" in particular you mean that this term can be used culturally/colloquially in recipes etc to cover other sibling genera like crayfish?  I'm warming to the shrimp (sensu X) class approach, e.g. "shrimp (sensu ingredient)", or "spinach (sensu ingredient)".  Am working on a way to best mark up such a class with statements like "has taxonomic reference" or "has regional usage".

**To all: I wanted to see if we agree that wherever possible, its best to avoid using synonyms (unless they reference scientific names, which seems ok.) and instead create "term (sensu ingredient)" classes that then relate the class to the things it is a synonym of/for, including logical equivalency if applicable?** It seems to me this offers more direct, stronger reasoning power, and also prevents us from loosing the information of when a particular synonym was used in a source database (i.e. through data import we have an id for "crawfish" rather than having it transformed into "crayfish" id.  However, is there substantial computational drawback though to having tonnes of logical equivalency statements?

**Comment by Public-Health-Bioinformatics** [comment]
On last point, in interests of keeping things simple for now, I won't push for any changes to use of oboInOwl synonyms.  Might return to this later.  But there may still be need for term (sensu ...) in some cases, esp. homonyms.

**Comment by Public-Health-Bioinformatics** [comment]
An update: I've just tweaked the LANGUAL_import.owl file, testing out the use of "inheres in" and "is bearer of" to make the link between animal and plant as food source, and NCBITaxon entries at the species/subspecies level.  You can see it at: https://github.com/Public-Health-Bioinformatics/foodon-langual .  Use the "inheres in" link to jump over to taxon, and "is bearer of" to jump back to organism as food role. The point is that particular uses of "Tomato" are by (social/local) convention pointing to particular biological species.  But looking to the group to say exactly how they think this relation should be expressed.

**Comment by Public-Health-Bioinformatics** [comment]
An update on the above:

1) We're using the Ancestry ancestro ontology to associate foods with ethnic origin/use.
 
2) I've retreated from the idea that a separate ontology ID will exist for each synonym of a term/concept.  This becomes too onerous in the service of trying to capture some recipe's original wording.  We'll leave it to a recipe database to preserve whichever synonyms were originally used to describe a recipe.

3) We will employ [term name (sensu qualifier)] wherever there is a need to disambiguate homonyms.

4) Currently the relation between a food source item and its taxonomy is:

- For a food item, we've translated a LanguaL reference to its taxonomic ITIS species ID as:
    [NCBITaxon name] and (has role some food)
The NCBITaxon name was looked up from the ITIS species ID by way of eol.org

- references from a food item to an ITIS family/genus/ other rank are handled as "has broad synonym" [NCBITaxon ID]

No alternate taxonomy lookup was performed although there is opportunity to do so manually or via scripts.  This will have to occur over time as resources permit.
